src/classes/VendorCommunicator.ts

```diff
@@ -14,111 +14,81 @@ export enum VendorEvent {
   UnauthorizedToken = 'UnauthorizedToken',
 }
 
-interface AddToCartMetadata {
-  source?: string;
-  title: string;
-  brand: string;
-  style: string;
-  color: string;
-  thumbnailUrl: string;
-  area?: number;
-  price?: number;
-  retailPrice?: number;
-}
-
-interface AddToCartPayload {
-  schema: string;
+interface CommonPayload {
```

Comment:
@NothingButThyme I took a pass at creating general/common interfaces that are extended for specific events. This commit is a proposal for us to discuss, not intended to unilaterally override the commit you pushed up (didn't think a whole separate PR for it was warranted). Thoughts appreciated!

The main point to get across here is the requirement for `token` and `customerId` be included in the majority of event payloads (excluding non-customer-centric events like tracking) so we can validate the token and make sure it's intended for the right customer.

Reply:
The changes look good. I don't have any objections.

I am curious what future-proofing is being accomplished by CommonMedata vs CabinetsMetadata, because 'source' and 'area' seem like they would apply to other things besides cabinets.